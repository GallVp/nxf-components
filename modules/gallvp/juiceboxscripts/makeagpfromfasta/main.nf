process JUICEBOXSCRIPTS_MAKEAGPFROMFASTA {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/juicebox_scripts:0.1.0gita7ae991--hdfd78af_0':
        'biocontainers/juicebox_scripts:0.1.0gita7ae991--hdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.agp")  , emit: agp
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "$meta.id"
    def VERSION = '0.1.0'
    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    makeAgpFromFasta.py \\
        $fasta \\
        "$prefix".agp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        juicebox_scripts: $VERSION
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "$meta.id"
    def VERSION = '0.1.0'
    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch "$prefix".agp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        juicebox_scripts: $VERSION
    END_VERSIONS
    """

}
