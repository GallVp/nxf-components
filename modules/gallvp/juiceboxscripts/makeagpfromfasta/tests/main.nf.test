nextflow_process {

    name "Test Process JUICEBOXSCRIPTS_MAKEAGPFROMFASTA"
    script "../main.nf"
    process "JUICEBOXSCRIPTS_MAKEAGPFROMFASTA"

    tag "modules"
    tag "modules_gallvp"
    tag "juiceboxscripts"
    tag "juiceboxscripts/makeagpfromfasta"

    test("sarscov2 - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - fasta  - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}