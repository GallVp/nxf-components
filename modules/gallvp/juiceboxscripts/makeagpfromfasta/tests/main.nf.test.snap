{
    "sarscov2 - fasta  - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.agp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,999a69e2270bed19914bb031bef66e12"
                ],
                "agp": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.agp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,999a69e2270bed19914bb031bef66e12"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-19T13:58:56.593979"
    },
    "sarscov2 - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.agp:md5,bb14c5d35ad337501d6a1aa51d4688cb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,999a69e2270bed19914bb031bef66e12"
                ],
                "agp": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.agp:md5,bb14c5d35ad337501d6a1aa51d4688cb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,999a69e2270bed19914bb031bef66e12"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-19T13:58:52.749741"
    }
}